### ISSUE TITLE
Mod added Villagers use the NORMAL Zombie texture

### ISSUE THREAD

**Issue by CovertJaguar** [MinecraftForge/MinecraftForge#3314]
Because of what appears to be an oversight in the code, non-vanilla villager zombies always get reset to normal zombies. I don't believe this is the desired behavior because everything else seems to support non-vanilla villager zombies.

The issue is the following line:
https://github.com/MinecraftForge/MinecraftForge/blob/1.10.x/src/main/java/net/minecraftforge/fml/common/registry/VillagerRegistry.java#L377

```
        if (network >= 0 && network < 5) // Vanilla
        {
            if (entity.getZombieType() == null || entity.getZombieType().getId() != network + 1)
            {
                entity.setZombieType(ZombieType.getVillagerByOrdinal(network));
            }
        }
        else if (entity.getZombieType() != null)
            entity.setZombieType(ZombieType.NORMAL);
```

`entity.getZombieType() != null` is always true because getZombieType() is the old vanilla function and always returns the 0 index in forge.

**Comment by Madsthunder** [comment]
Superseded by #3678

**Comment by bs2609** [comment]
This can be closed now (93a4e13837cc4139fd33f1f263c71ab8eeb4d82a)
